src/Interpreters/Context.cpp

```diff
@@ -1229,14 +1232,43 @@ void Context::addQueryFactoriesInfo(QueryLogFactories factory_type, const String
 
 StoragePtr Context::executeTableFunction(const ASTPtr & table_expression)
 {
+    const ASTFunction * function = assert_cast<const ASTFunction *>(table_expression.get());
+    String database_name = getCurrentDatabase();
+    String table_name = function->name;
+
+    if (function->has_database_name)
+    {
+        std::vector<std::string> parts;
+        splitInto<'.'>(parts, function->name);
+
+        if (parts.size() == 2)
+        {
+            database_name = parts[0];
+            table_name = parts[1];
+        }
+    }
+
+    StoragePtr table = DatabaseCatalog::instance().tryGetTable({database_name, table_name}, getQueryContext());
+    if (table)
+    {
+        if (table.get()->isView() && table->as<StorageView>()->isParameterizedView())
+            return table;
+    }
     auto hash = table_expression->getTreeHash();
     String key = toString(hash.first) + '_' + toString(hash.second);
-
     StoragePtr & res = table_function_results[key];
-
     if (!res)
     {
-        TableFunctionPtr table_function_ptr = TableFunctionFactory::instance().get(table_expression, shared_from_this());
+        TableFunctionPtr table_function_ptr;
+        try
+        {
+            table_function_ptr = TableFunctionFactory::instance().get(table_expression, shared_from_this());
+        }
+        catch (Exception & e)
+        {
+            e.addMessage(" or incorrect parameterized view");
```

Comment:
If we currently don't have a better idea, then let's keep this as it is, and have an improvement PR to pick this up.
In my opinion, adding this message gives the user information that if they are trying to use parameterized view, then the usage is incorrect instead of just getting `unknonwn function` error.